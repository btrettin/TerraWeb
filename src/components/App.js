import React, { Component } from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducers from "../reducers/reducers";
import { CSSTransition, TransitionGroup } from "";

import Error404 from "../components/errors/Error404";
import HomePageComponent from "./pages/HomePage";
import requestPipelineComponent from "./pages/requestPipeline";

export default class App extends Component {
  constructor() {
    super();

    const store = createStore(reducers);

    this.state = {
      store
    };

    store.subscribe(() => {
      console.log("store.getState()", store.getState());
    });
  }
  render() {
    return (
      <Provider store={this.state.store}>
        <BrowserRouter className="router">
          <Switch>
            <Route exact path="/test" component={App} />
            <Route exact path="/HomePage" component={HomePageComponent} />
            <Route
              exact
              path="/requestPipeline"
              component={requestPipelineComponent}
            />
            <Route path="/*" component={Error404} />
          </Switch>
        </BrowserRouter>
      </Provider>
    );
  }
}
