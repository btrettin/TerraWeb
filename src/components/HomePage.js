import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Menu, Modal, Input, Grid, Button, Form, Card } from 'semantic-ui-react';

import Styles from '../styles/HomePage';
import Actions from '../reducers/Actions'
import { statekeys } from '../helpers/Common' 

export class HomePage extends Component {
    render() {
        const {
            setEmail,
            setPassword,
            setFirstName,
            setLastName,
            setPasswordRepeat,
            setAddress,
            setCity,
            setAddressState,
            setZipcode,
            openLoginModal,
            closeLoginModal,
            isLoginModalOpen,
            openSignupModal,
            closeSignupModal,
            isSignupModalOpen
        } = this.props;

        return (
            <div>
                <div style={Styles.navbarContainer}>
                    <Menu secondary style={Styles.loginSignupMenu}>
                        <Menu.Menu position='right'>
                            <Modal
                                style={Styles.modal}
                                size='mini'
                                trigger={
                                    <Menu.Item style={Styles.menuItem} onClick={openLoginModal}>
                                        Login
                                    </Menu.Item>
                                }
                                open={isLoginModalOpen}
                                onClose={closeLoginModal}
                                closeIcon
                            >
                                <Modal.Header style={Styles.modalHeader}>Welcome Back!</Modal.Header>
                                <Modal.Description>
                                    <Grid style={Styles.loginGrid}>
                                        <Grid.Row>
                                            <Grid.Column>
                                                <Input
                                                    fluid
                                                    icon='user'
                                                    iconPosition='left'
                                                    placeholder='someone@example.com'
                                                    onChange={setEmail}
                                                />
                                            </Grid.Column>
                                        </Grid.Row>
                                        <Grid.Row>
                                            <Grid.Column>
                                                <Input
                                                    fluid
                                                    icon='lock'
                                                    type='password'
                                                    iconPosition='left'
                                                    placeholder='Password'
                                                    onChange={setPassword}
                                                />
                                            </Grid.Column>
                                        </Grid.Row>
                                        <Grid.Row centered>
                                            <Grid.Column width={6}>
                                                <Button
                                                    style={Styles.modalButton}
                                                    size='large'
                                                    onClick={closeLoginModal}
                                                >
                                                    Login
                                                </Button>
                                            </Grid.Column>
                                        </Grid.Row>
                                    </Grid>
                                </Modal.Description>
                            </Modal>
                            <Modal
                                style={Styles.modal}
                                size='tiny'
                                trigger={
                                    <Menu.Item style={Styles.menuItem} onClick={openSignupModal}>
                                    Sign Up
                                    </Menu.Item>
                                }
                                open={isSignupModalOpen}
                                onClose={closeSignupModal}
                                closeIcon
                            >
                                <Modal.Header style={Styles.modalHeader}>Sign Up!</Modal.Header>
                                <Modal.Description>
                                    <Form style={Styles.signUpForm}>
                                        <Form.Group widths='equal'>
                                            <Form.Input
                                                fluid
                                                label='First Name'
                                                color='white'
                                                placeholder='First Name'
                                                onChange={setFirstName}
                                            />
                                            <Form.Input
                                                fluid
                                                label='Last Name'
                                                placeholder='Last Name'
                                                onChange={setLastName}
                                            />
                                        </Form.Group>
                                        <Form.Group widths='equal'>
                                            <Form.Input
                                                fluid
                                                label='Email'
                                                placeholder='someone@example.com'
                                                onChange={setEmail}
                                            />
                                        </Form.Group>
                                        <Form.Group widths='equal'>
                                            <Form.Input
                                                fluid
                                                label='Password'
                                                type='password'
                                                placeholder='Password'
                                                onChange={setPassword}
                                            />
                                            <Form.Input
                                                fluid
                                                label='Repeat Password'
                                                type='password'
                                                placeholder='Repeat Password'
                                                onChange={setPasswordRepeat}
                                            />
                                        </Form.Group>
                                        <Form.Group widths='equal'>
                                            <Form.Input
                                                fluid
                                                label='Address'
                                                placeholder='Address'
                                                onChange={setAddress}
                                            />
                                            <Form.Input
                                                fluid
                                                label='City'
                                                placeholder='City'
                                                onChange={setCity}
                                            />
                                        </Form.Group>
                                        <Form.Group widths='equal'>
                                            <Form.Dropdown
                                                fluid
                                                label='State'
                                                placeholder='State'
                                                search
                                                selection
                                                options={statekeys}
                                                onChange={setAddressState}
                                            />
                                            <Form.Input
                                                fluid
                                                label='Zipcode'
                                                placeholder='Zipcode'
                                                onChange={setZipcode}
                                            />
                                        </Form.Group>
                                        <Form.Group style={Styles.signUpCheckbox}>
                                            <Form.Checkbox label='I agree to the Terms and Conditions'/>
                                        </Form.Group>
                                        <Form.Group>
                                            <Form.Button
                                                style={Styles.modalButton}
                                                size='large'
                                                onClick={closeSignupModal}
                                            >
                                                Submit
                                            </Form.Button>
                                        </Form.Group>
                                    </Form>
                                </Modal.Description>
                            </Modal>
                        </Menu.Menu>
                    </Menu>
                </div>
                <div style={Styles.contentContainer}>
                    <div style={Styles.cardContainer}>
                        <Card fluid>
                            <Card.Content>
                                <Card.Header>
                                    Tell us how you want your yard cut...
                                </Card.Header>
                            </Card.Content>
                            <Card.Content>
                                <Grid centered style={Styles.cardContent}>
                                    <Grid.Row>
                                        <Grid.Column>
                                            <div style={Styles.cardInputContainer}>
                                                <span style={Styles.cardInputLabel}>Yard Size:</span>
                                                <Input label={{basic: true, content: 'ft²'}} labelPosition="right"/>
                                            </div>
                                        </Grid.Column>
                                    </Grid.Row>
                                    <Grid.Row>
                                        <Grid.Column>
                                            <div style={Styles.cardInputContainer}>
                                                <span style={Styles.cardInputLabel}>Grass Height:</span>
                                                <Input label={{basic: true, content: 'in.'}} labelPosition="right"/>
                                            </div>
                                        </Grid.Column>
                                    </Grid.Row>
                                    <Grid.Row>
                                        <Grid.Column>
                                            <div style={Styles.cardButtonContainer}>
                                                <Button
                                                    style={Styles.modalButton}
                                                    size='large'
                                                >
                                                    Continue
                                                </Button>
                                            </div>
                                        </Grid.Column>
                                    </Grid.Row>
                                </Grid>
                            </Card.Content>
                        </Card>
                    </div>
                </div>
                <div style={Styles.footerContainer}>
                    <div style={Styles.footerContentContainer}>
                        <span>About</span>
                        <span>Contact</span>
                        <span>FAQ</span>
                        <span>Services</span>
                    </div>
                    <div style={Styles.footerCopyright}>©2018 Terra Development, Inc.</div>
                </div>
            </div>
        );
    }
};

const mapDispatchToProps = dispatch => {
    return {
        setEmail: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setEmail,
                    value
                };

            dispatch(action);
        },
        setPassword: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setPassword,
                    value
                };

            dispatch(action);
        },
        setFirstName: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setFirstName,
                    value
                };

            dispatch(action);
        },
        setLastName: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setLastName,
                    value
                };

            dispatch(action);
        },
        setPasswordRepeat: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setPasswordRepeat,
                    value
                };

            dispatch(action);
        },
        setAddress: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setAddress,
                    value
                };

            dispatch(action);
        },
        setCity: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setCity,
                    value
                };

            dispatch(action);
        },
        setAddressState: (event,data) => {
            const value = data.value,
                action = {
                    type: Actions.login.setAddressState,
                    value
                };

            dispatch(action);
        },
        setZipcode: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setZipcode,
                    value
                };

            dispatch(action);
        },
        openLoginModal: () => {
            const action = {
                type: Actions.login.toggleLoginModal
            };

            dispatch(action);
        },
        closeLoginModal: () => {
            const action = {
                type: Actions.login.clear
            };

            dispatch(action);
        },
         openSignupModal: () => {
            const action = {
                type: Actions.login.toggleSignupModal
            };

            dispatch(action);
        },
        closeSignupModal: () => {
            const action = {
                type: Actions.login.clear
            };

            dispatch(action);
        }
    };
}

export default connect(() => { return {} }, mapDispatchToProps)(HomePage);
