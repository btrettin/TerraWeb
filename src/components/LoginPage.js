import React, { Component } from 'react';
import { connect } from 'react-redux';

import FormInput from './helpers/FormInput';
import FormInputList from './helpers/FormInputList';

import ListHelper from '../helpers/ListHelper';

import '../sass/LoginPage.css';

const Actions = require('../reducers/Actions')();

export class LoginPage extends Component {
    constructor() {
        super();
        this.state = {
            tabSelected: "login"
        };
    }

    chooseTab(tab) {
        const {clearPassword} = this.props;

        this.setState(previousState => {
            if (previousState.tabSelected !== tab)
                clearPassword();

            return Object.assign({}, previousState, {
                tabSelected: tab
            });
        });
    }

    tabIsSelected(tab) {
        if (tab === this.state.tabSelected)
            return "login-tab--selected"
        
        return;
    }

    renderForm() {
        const {
            firstName,
            lastName,
            email,
            password,
            passwordRepeat,
            address,
            city,
            addressState,
            zipcode,
            driversLiscenseNumber,
            setEmail,
            setPassword,
            setFirstName,
            setLastName,
            setPasswordRepeat,
            setAddress,
            setCity,
            setAddressState,
            setZipcode,
            setDriversLiscenseNumber
        } = this.props;

        if (this.state.tabSelected === 'login') 
            return (
                <div>
                    <FormInput
                        label="Email"
                        className="login-body-email"
                        value={email}
                        onChange={setEmail}
                    />
                    <FormInput
                        label="Password"
                        className="login-body-password"
                        type="password"
                        value={password}
                        onChange={setPassword}
                    />
                    <div className="login-button-container">
                        <div className="login-button">LOGIN</div>
                    </div>
                </div>
            );
        
        return (
            <div>
                <div className="login-signup-form-container">
                    <FormInput
                        label="First Name"
                        className="login-signup-form-input"
                        value={firstName}
                        onChange={setFirstName}
                    />
                    <div className="login-signup-form-spacer"></div>
                    <FormInput
                        label="Last Name"
                        className="login-signup-form-input"
                        value={lastName}
                        onChange={setLastName}
                    />
                </div>
                <div className="login-signup-form-container">
                    <FormInput
                        label="Email Address"
                        className="login-signup-form-input"
                        value={email}
                        onChange={setEmail}
                    />
                </div>
                <div className="login-signup-form-container">
                    <FormInput
                        label="Password"
                        type="password"
                        className="login-signup-form-input"
                        value={password}
                        onChange={setPassword}
                    />
                    <div className="login-signup-form-spacer"></div>                    
                    <FormInput
                        label="Repeat Password"
                        type="password"
                        className="login-signup-form-input"
                        value={passwordRepeat}
                        onChange={setPasswordRepeat}
                    />
                </div>
                <div className="login-signup-form-container">
                    <FormInput
                        label="Address"
                        className="login-signup-form-input"
                        value={address}
                        onChange={setAddress}
                    />
                </div>
                <div className="login-signup-form-container">
                    <FormInput
                        label="City"
                        className="login-signup-form-input"
                        value={city}
                        onChange={setCity}
                    />
                    <div className="login-signup-form-spacer"></div>                    
                    <FormInputList
                        label="State"
                        className="login-signup-form-input"
                        value={addressState}
                        onChange={setAddressState}
                        options={ListHelper.getStates()}
                    />
                    <div className="login-signup-form-spacer"></div>                    
                    <FormInput
                        label="Zipcode"
                        className="login-signup-form-input"
                        value={zipcode}
                        onChange={setZipcode}
                    />
                </div>
                <div className="login-signup-form-container">
                    <FormInput
                        label="Driver's Liscence Number"
                        className="login-signup-form-input"
                        value={driversLiscenseNumber}
                        onChange={setDriversLiscenseNumber}
                    />
                </div>
                <div className="login-signup-button-container">
                    <div className="login-button">SIGN UP!</div>
                </div>
            </div>
        )
    }

    render() {
        return (
            <div className="login-page-container">
                <div className="login-skew"></div>
                <div className="login-tab-container">
                    <div 
                        className={`login-tab login-tab--left ${this.tabIsSelected("login")}`}
                        onClick={this.chooseTab.bind(this, "login")}
                    >
                        Login
                    </div>
                    <div
                        className={`login-tab login-tab--right ${this.tabIsSelected("signup")}`}
                        onClick={this.chooseTab.bind(this, "signup")}
                    >
                        Sign Up
                    </div>
                </div>
                <div className="login-body">
                    {this.renderForm()}
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => {
    const {
        firstName,
        lastName,
        email,
        password,
        passwordRepeat,
        address,
        city,
        addressState,
        zipcode,
        driversLiscenseNumber
    } = state.login;

    return {
        firstName,
        lastName,
        email,
        password,
        passwordRepeat,
        address,
        city,
        addressState,
        zipcode,
        driversLiscenseNumber
    };
};

const mapDispatchToProps = dispatch => {
    return {
        setEmail: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setEmail,
                    value
                };

            console.log(Actions);

            dispatch(action);
        },
        setPassword: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setPassword,
                    value
                };

            dispatch(action);
        },
        setFirstName: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setFirstName,
                    value
                };

            dispatch(action);
        },
        setLastName: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setLastName,
                    value
                };

            dispatch(action);
        },
        setPasswordRepeat: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setPasswordRepeat,
                    value
                };

            dispatch(action);
        },
        setAddress: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setAddress,
                    value
                };

            dispatch(action);
        },
        setCity: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setCity,
                    value
                };

            dispatch(action);
        },
        setAddressState: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setAddressState,
                    value
                };

            dispatch(action);
        },
        setZipcode: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setZipcode,
                    value
                };

            dispatch(action);
        },
        setDriversLiscenseNumber: event => {
            const value = event.target.value,
                action = {
                    type: Actions.login.setDriversLiscenseNumber,
                    value
                };

            dispatch(action);
        },
        clearPassword: () => {
            const action = {
                type: Actions.login.clearPassword
            };

            dispatch(action);
        }
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
