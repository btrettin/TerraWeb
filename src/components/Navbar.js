import React, { Component } from 'react';

import '../sass/Navbar.css';

export default class Navbar extends Component {
    render() {
        return (
            <div className="navbar-container">
                <div className="navbar-home-title">
                    <a 
                        href="/login"
                        className="navbar-home-link"
                    >
                        MyRide
                    </a>
                </div>
            </div>
        );
    }
}
