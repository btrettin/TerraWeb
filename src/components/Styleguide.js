import React, { Component } from 'react';

import '../sass/Styleguide.css';

export default class Styleguide extends Component {
    render() {
        return (
            <div className="styleguide-container">
                <div className="styleguide-content-spacer"></div>
                <div className="styleguide-content-container">
                    <h1>MyRide Styleguide</h1>
                    <hr/>
                    <div className="styleguide-content-colors">
                        <h2>Colors:</h2>
                        <p>
                            Below are the colors that should be used in the
                            main content of the website.
                        </p>
                        <div className="styleguide-content-colors-demo"></div>
                    </div>
                </div>
            </div>
        )
    };
}
