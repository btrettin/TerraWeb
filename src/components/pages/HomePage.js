import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Menu, Modal, Input, Grid, Button, Form } from 'semantic-ui-react';
import { Link } from 'react-router-dom'

import requestPipeline from '../../components/pages/requestPipeline';

import Styles from '../../styles/HomePage';
import Actions from '../../reducers/Actions'
import { statekeys } from '../../helpers/Common'

export class HomePage extends Component {
  render() {
    const {setEmail, setPassword, setFirstName, setLastName, setPasswordRepeat, setAddress, setCity, setAddressState, setZipcode, openLoginModal, closeLoginModal, isLoginModalOpen, openSignupModal, closeSignupModal, isSignupModalOpen} = this.props;

    return (
            <div>
            <div style={Styles.navbarContainer}>
                <Menu secondary style={Styles.loginSignupMenu}>
                    <Menu.Menu position='right'>
                        <Modal
      style={Styles.modal}
      size='mini'
      trigger={
      <Menu.Item style={Styles.menuItem} onClick={openLoginModal}>
                                    Login
                                </Menu.Item>
      }
      open={isLoginModalOpen}
      onClose={closeLoginModal}
      closeIcon
      >
                            <Modal.Header style={Styles.modalHeader}>Welcome Back!</Modal.Header>
                            <Modal.Description>
                                <Grid style={Styles.loginGrid}>
                                    <Grid.Row>
                                        <Grid.Column>
                                            <Input
      fluid
      icon='user'
      iconPosition='left'
      placeholder='someone@example.com'
      onChange={setEmail}
      />
                                        </Grid.Column>
                                    </Grid.Row>
                                    <Grid.Row>
                                        <Grid.Column>
                                            <Input
      fluid
      icon='lock'
      type='password'
      iconPosition='left'
      placeholder='Password'
      onChange={setPassword}
      />
                                        </Grid.Column>
                                    </Grid.Row>
                                    <Grid.Row centered>
                                        <Grid.Column width={6}>
                                            <Button
      style={Styles.modalButton}
      size='large'
      onClick={closeLoginModal}
      >
                                                Login
                                            </Button>
                                        </Grid.Column>
                                    </Grid.Row>
                                </Grid>
                            </Modal.Description>
                        </Modal>
                        <Modal
      style={Styles.modal}
      size='tiny'
      trigger={
      <Menu.Item style={Styles.menuItem} onClick={openSignupModal}>
                                Sign Up
                                </Menu.Item>
      }
      open={isSignupModalOpen}
      onClose={closeSignupModal}
      closeIcon
      >
                            <Modal.Header style={Styles.modalHeader}>Sign Up!</Modal.Header>
                            <Modal.Description>
                                <Form style={Styles.signUpForm}>
                                    <Form.Group widths='equal'>
                                        <Form.Input
      fluid
      label='First Name'
      color='white'
      placeholder='First Name'
      onChange={setFirstName}
      />
                                        <Form.Input
      fluid
      label='Last Name'
      placeholder='Last Name'
      onChange={setLastName}
      />
                                    </Form.Group>
                                    <Form.Group widths='equal'>
                                        <Form.Input
      fluid
      label='Email'
      placeholder='someone@example.com'
      onChange={setEmail}
      />
                                    </Form.Group>
                                    <Form.Group widths='equal'>
                                        <Form.Input
      fluid
      label='Password'
      type='password'
      placeholder='Password'
      onChange={setPassword}
      />
                                        <Form.Input
      fluid
      label='Repeat Password'
      type='password'
      placeholder='Repeat Password'
      onChange={setPasswordRepeat}
      />
                                    </Form.Group>
                                    <Form.Group widths='equal'>
                                        <Form.Input
      fluid
      label='Address'
      placeholder='Address'
      onChange={setAddress}
      />
                                        <Form.Input
      fluid
      label='City'
      placeholder='City'
      onChange={setCity}
      />
                                    </Form.Group>
                                    <Form.Group widths='equal'>
                                        <Form.Dropdown
      fluid
      label='State'
      placeholder='State'
      search
      selection
      options={statekeys}
      onChange={setAddressState}
      />
                                        <Form.Input
      fluid
      label='Zipcode'
      placeholder='Zipcode'
      onChange={setZipcode}
      />
                                    </Form.Group>
                                    <Form.Group style={Styles.signUpCheckbox}>
                                        <Form.Checkbox label='I agree to the Terms and Conditions'/>
                                    </Form.Group>
                                    <Form.Group>
                                        <Form.Button
      style={Styles.modalButton}
      size='large'
      onClick={closeSignupModal}
      >
                                            Submit
                                        </Form.Button>
                                    </Form.Group>
                                </Form>
                            </Modal.Description>
                        </Modal>
                    </Menu.Menu>
                </Menu>
            </div>
            <Grid style={Styles.header} centered padded relaxed>
            <Grid.Row>
            </Grid.Row>
             <Grid.Row>
             <Link to='/requestPipeline'>
             <Button size='large'>Make a Request</Button>
             </Link>
            </Grid.Row>

             <Grid.Row>
            </Grid.Row>

            </Grid>
            </div>
    );
  }
}
;

const mapDispatchToProps = dispatch => {
  return {
    setEmail: event => {
      const value = event.target.value;
      const action = {
          type: Actions.login.setEmail,
          value
        };

      dispatch(action);
    },
    setPassword: event => {
      const value = event.target.value;
      const action = {
          type: Actions.login.setPassword,
          value
        };

      dispatch(action);
    },
    setFirstName: event => {
      const value = event.target.value;
      const action = {
          type: Actions.login.setFirstName,
          value
        };

      dispatch(action);
    },
    setLastName: event => {
      const value = event.target.value;
      const action = {
          type: Actions.login.setLastName,
          value
        };

      dispatch(action);
    },
    setPasswordRepeat: event => {
      const value = event.target.value;
      const action = {
          type: Actions.login.setPasswordRepeat,
          value
        };

      dispatch(action);
    },
    setAddress: event => {
      const value = event.target.value;
      const action = {
          type: Actions.login.setAddress,
          value
        };

      dispatch(action);
    },
    setCity: event => {
      const value = event.target.value;
      const action = {
          type: Actions.login.setCity,
          value
        };

      dispatch(action);
    },
    setAddressState: (event, data) => {
      const value = data.value;
      const action = {
          type: Actions.login.setAddressState,
          value
        };

      dispatch(action);
    },
    setZipcode: event => {
      const value = event.target.value;
      const action = {
          type: Actions.login.setZipcode,
          value
        };

      dispatch(action);
    },
    openLoginModal: () => {
      const action = {
        type: Actions.login.toggleLoginModal
      };

      dispatch(action);
    },
    closeLoginModal: () => {
      const action = {
        type: Actions.login.clear
      };

      dispatch(action);
    },
    openSignupModal: () => {
      const action = {
        type: Actions.login.toggleSignupModal
      };

      dispatch(action);
    },
    closeSignupModal: () => {
      const action = {
        type: Actions.login.clear
      };

      dispatch(action);
    }
  };
}

export default connect(() => {
  return {}
}, mapDispatchToProps)(HomePage);
