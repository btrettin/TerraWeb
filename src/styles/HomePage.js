const green = '#63BA18';
const white = '#FFFFFF'

export default {
    navbarContainer: {
        width: '100%',
        minHeight: '3.5vh',
        backgroundColor: green,
        position: 'fixed',
        top: '0px'
    },
    loginSignupMenu: {
        paddingRight: '1rem'
    },
    menuItem: {
        color: 'white'
    },
    modal: {
        backgroundColor: 'white',
        padding: '1rem'
    },
    loginGrid: {
        margin: '1rem'
    },
    modalHeader: {
        display: 'flex',
        justifyContent: 'center',
        color: green,
        fontSize:'2rem',
        backgroundColor: white,
        borderBottom: 'none'
    },
    modalButton: {
        backgroundColor: green,
        color: 'white'
    },
    signUpForm: {
        padding: '1rem'
    },
    signUpCheckbox: {
        paddingBottom: '.5rem'
    },
    contentContainer: {
        paddingTop: '3.5vh',
        height: '96.5vh',
        backgroundImage: 'url("http://www.tokkoro.com/picsup/3004995-blur_blurred_eco_environment_garden_grass_green_lawn_mowing_whitespace_yard.jpg")',
        backgroundSize: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center'
    },
    cardContainer: {
        width: '20%',
        minWidth: '300px'
    },
    cardInputContainer: {
        display: 'flex',
        justifyContent: 'space-between',
    },
    cardInputLabel: {
        paddingTop: '9.5px',
        fontWeight: 'bold'
    },
    cardContent: {
        paddingTop: '2vh'
    },
    cardButtonContainer: {
        display: 'flex',
        justifyContent: 'center',
        paddingBottom: '2vh'
    },
    footerContainer: {
        backgroundColor: '#CECECE'
    },
    footerContentContainer: {
        paddingTop: '2vh',
        display: 'flex',
        justifyContent: 'space-around',
        width: '50vw',
        margin: '0 auto'
    },
    footerCopyright: {
        textAlign: 'center',
        color: '#AAAAAA',
        padding: '2vh 0'
    }
};
