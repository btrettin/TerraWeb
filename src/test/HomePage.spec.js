import React from 'react';
import { HomePage } from '../components/HomePage';
import renderer from 'react-test-renderer';

test('Home page looks correct visually', () => {
    const component = renderer.create(<HomePage/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
