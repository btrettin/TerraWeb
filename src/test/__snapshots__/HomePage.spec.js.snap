// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page looks correct visually 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#63BA18",
      "minHeight": 42,
      "width": "100%",
    }
  }
>
  <div
    className="ui secondary menu"
    style={
      Object {
        "paddingRight": "1rem",
      }
    }
  >
    <div
      className="right menu"
    >
      <a
        className="item"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "white",
          }
        }
      >
        Login
      </a>
      <a
        className="item"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "white",
          }
        }
      >
        Sign Up
      </a>
    </div>
  </div>
</div>
`;
